Framework/src/CheckRunner.cxx

```diff
@@ -181,11 +180,17 @@ void CheckRunner::refreshConfig(InitContext& iCtx)
       // prepare the information we need
       auto infrastructureSpec = InfrastructureSpecReader::readInfrastructureSpec(updatedTree);
 
-      // TODO: use the config to reconfigure the check runner.
-      // TODO: in particular, reset mChecks and update it.
-      // TODO: Problem is that a lot of the logic is in the infrastructure generator.
-      // TODO: we should probably just preserve the checks list and update their state.
-      // TODO: also see if we should update the detector name
+      // Use the config to reconfigure the check runner.
+      // The configs for the checks we find in the config and in our map are updated.
+      // Topology changes are ignored: New checks are ignored. Removed checks are ignored.
+      for (const auto& checkSpec : infrastructureSpec.checks) {
+        // search if we have this check in this runner and replace it
+        if ( mChecks.find(checkSpec.checkName) != mChecks.end() ) {
+          auto checkConfig = Check::extractConfig(infrastructureSpec.common, checkSpec);
+          mChecks[checkSpec.checkName] = checkConfig;
```

Comment:
I am surprised that a Check constructor for checkConfig can be used this way, but if that compiles then I have nothing against.

Reply:
To be honest I was surprised myself :) but this was already the way it was done in the constructor.